boards/cats_rev1Pro/src/control/data_processing.c

```diff
@@ -20,82 +20,78 @@
 #include <string.h>
 #include <stdint.h>
 
-// Utility function to find minimum of two integers
-int min(int x, int y) { return (x < y) ? x : y; }
+#define SWAP(a,b) { float temp=(a);(a)=(b);(b)=temp; }
+#define SORT_TWO(a,b) { if ((a)>(b)) SWAP((a),(b)); }
 
-/* Function to merge the two haves arr[l..m] and arr[m+1..r] of array arr[] */
-void merge(float arr[], int l, int m, int r) {
-  int i, j, k;
-  int n1 = m - l + 1;
-  int n2 = r - m;
+float median(float input_array[], int32_t array_size) {
+  float array[array_size];
```

Comment:
Declaring an array inside a function of size `N` must be determined in compile-time. I just checked and saw that we always did it like that... this probably worked because we have a constant in the function call but we should fix that now. Please use the argument only to check if the function was called with what we expect and put a constant for the array size e.g. ` float array[9]`

Reply:
There's different ways I can fix this.

Option (1) is just setting the size to 9 and doing the hardwired algorithm. It would make sense to put an assert for checking the argument but I haven't seen any in the codebase, and I guess there's a reason why you don't use asserts. Option (2) is setting the size to MEDIAN_FILTER_SIZE which is a constant and then doing having the if-else so we can have the two algorithms in case we change the filter size.

I think option (2) is the best choice so I'm gonna go with it. In any case I don't know what to do with array_size since I think we don't have any asserts, we could totally just get rid of the argument. And I don't know whether I should put the if-else as a compile-time statement. I'm just gonna act as if it was my own codebase and do what I consider best for keeping it efficient and easy to understand, in that order.

Reply:
Sounds good, I would probably recommend doing something like this to keep the option open to change the median filter size:

```
#if MEDIAN_FILTER_SIZE  == 9
float median(float input_array[]) {
// Hardwired algorithm
}
#else
float median(float input_array[]) {
float array[MEDIAN_FILTER_SIZE];
// Dynamic array size
}
#endif
```